internals/daemon/api_checks.go

```diff
@@ -66,3 +67,34 @@ func v1GetChecks(c *Command, r *http.Request, _ *UserState) Response {
 	}
 	return SyncResponse(infos)
 }
+
+func v1PostChecks(c *Command, r *http.Request, _ *UserState) Response {
+	var payload struct {
+		Action string `json:"action"`
+		Check  string `json:"check"`
+	}
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return BadRequest("cannot decode data from request body: %v", err)
+	}
+	switch payload.Action {
+	case "run":
+		if payload.Check == "" {
+			return BadRequest("no check to %s provided", payload.Action)
+		}
+
+		check, ok := c.d.overlord.PlanManager().Plan().Checks[payload.Check]
+		if !ok {
+			return SyncResponse("check not found")
+		}
+
+		err := c.d.overlord.CheckManager().RunCheck(r.Context(), check)
+		if err != nil {
+			return SyncResponse(err)
+		}
+		return SyncResponse("check passed")
+
+	default:
+		return BadRequest("action %s not supported", payload.Action)
```

Comment:
In other places we use `"invalid action %q"` -- let's just be careful to be consistent for things like this.

Reply:
Hmm, I used this wording too, because we already use it for services and changes:

```
tameyer@tam-canoncial-1:~/code/pebble$ rg 'is unsupported'
internals/daemon/api_changes.go
246:		return BadRequest("change action %q is unsupported", reqData.Action)

internals/daemon/api_checks.go
106:		return BadRequest("action %q is unsupported", payload.Action)

internals/daemon/api_services.go
185:		return BadRequest("action %q is unsupported", payload.Action)
```

Reply:
Hmm, true -- thanks for the detailed info. I guess we should pick one. I'd vote for `invalid action %q` as it's a bit more common, and "invalid" seems more right than "unsupported" to me (that implies it might be supported on another system/version).

@IronCore864, can you please change the "unsupported" ones to `invalid action %q`?